core/src/main/java/org/wso2/msf4j/internal/deployer/MicroservicesDeployer.java

```diff
@@ -0,0 +1,279 @@
+                ).findAny().isPresent();
+    }
+
+    /**
+     * Returns the extension of the artifact file.
+     * @param file artifact file
+     * @return file extension
+     */
+    private String getFileExtension(File file) {
+        String extension = "";
+        if (file == null) {
+            return extension;
+        }
+
+        String fileName = file.getName();
+        if (file.isFile()) {
+            int i = fileName.lastIndexOf('.');
+            if (i > 0) {
+                extension = fileName.substring(i + 1);
+            }
+        }
+        return extension;
+    }
+
+    /**
+     * Add micro services instance to micro services registry.
+     * @param service object to be added to registry
+     */
+    private boolean addService(Microservice service) {
+        Map<String, MicroservicesRegistryImpl> microservicesRegistries =
+                org.wso2.msf4j.internal.DataHolder.getInstance().getMicroservicesRegistries();
+        if (microservicesRegistries.isEmpty()) {
+            log.error("Microservice deployment failed. Microservices Registry doesn't exist to register microservice.");
+            return false;
+        }
+        microservicesRegistries.values().forEach(registry -> registry.addService(service));
+        log.info("Microservice deployed successfully");
+        return true;
+    }
+
+    /**
+     * Remove micro services instance to micro services registry.
+     * @param service object to be removed from registry,
+     */
+    private boolean removeService(Microservice service) {
+        Map<String, MicroservicesRegistryImpl> microservicesRegistries =
+                org.wso2.msf4j.internal.DataHolder.getInstance().getMicroservicesRegistries();
+        if (microservicesRegistries.isEmpty()) {
+            log.error("Microservice removal failed. Microservices Registry doesn't exist to register microservice.");
+            return false;
+        }
+
+        microservicesRegistries.values().forEach(registry -> {
+            registry.removeService(service.getClass()
+                    .getAnnotation(Path.class).value());
+            registry.preDestroyService(service);
+        });
+        log.info("Microservice undeployed successfully");
+        return true;
+    }
+
+    @Override
+    public void onAllRequiredCapabilitiesAvailable() {
+        DataHolder.getInstance().getBundleContext().registerService(MicroservicesDeployer.class, this, null);
+        if (log.isDebugEnabled()) {
+            log.debug("MicroservicesDeployer service is available");
+        }
+    }
+
+    @Reference(
+            name = "microservices-server",
+            service = MicroservicesServerSC.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "removeMicroservicesServer"
+    )
+    protected void addMicroservicesServer(MicroservicesServerSC microservicesServerSC) {
```

Comment:
Add method comment to describe the usage. commit revision 2e8f63cb6a69e94ef91196d66631f9ec7733b099